vuu-ui/packages/vuu-shell/src/layout-management/useLayoutManager.tsx

```diff
@@ -1,46 +1,69 @@
 import React, { useState, useCallback, useContext, useEffect } from "react";
-import { getLocalEntity } from "@finos/vuu-filters";
 import { LayoutJSON, LocalLayoutPersistenceManager } from "@finos/vuu-layout";
 import { LayoutMetadata } from "./layoutTypes";
+import { defaultLayout } from "@finos/vuu-layout/";
 
 const persistenceManager = new LocalLayoutPersistenceManager();
 
 export const LayoutManagementContext = React.createContext<{
   layoutMetadata: LayoutMetadata[],
-  saveLayout: (n: Omit<LayoutMetadata, "id">) => void
-}>({ layoutMetadata: [], saveLayout: () => { } })
+  saveLayout: (n: Omit<LayoutMetadata, "id">) => void,
+  currentLayout: LayoutJSON,
+  saveCurrentLayout: (layout: LayoutJSON) => void,
+  loadLayoutById: (id: string) => void
+}>({
+  layoutMetadata: [],
+  saveLayout: () => { },
+  currentLayout: defaultLayout,
+  saveCurrentLayout: () => { },
+  loadLayoutById: () => defaultLayout
+})
 
-export const LayoutManagementProvider = (props: {
-    children: JSX.Element | JSX.Element[]
-  }) => {
+type LayoutManagementProviderProps = {
+  children: JSX.Element | JSX.Element[]
+}
+
+export const LayoutManagementProvider = (props: LayoutManagementProviderProps) => {
   const [layoutMetadata, setLayoutMetadata] = useState<LayoutMetadata[]>([]);
+  const [currentLayout, setCurrentLayout] = useState<LayoutJSON>(defaultLayout);
 
   useEffect(() => {
-    const loadedMetadata = persistenceManager.loadMetadata();
-    setLayoutMetadata(loadedMetadata || [])
+    persistenceManager.loadMetadata().then(metadata => {
+      setLayoutMetadata(metadata)
+    })
+    persistenceManager.loadCurrentLayout().then(layout => {
+      setCurrentLayout(layout);
+    })
   }, [])
 
-  const saveLayout = useCallback((metadata: Omit<LayoutMetadata, "id">) => {
-    const json = getLocalEntity<LayoutJSON>("api/vui");
+  const saveCurrentLayout = useCallback((layout: LayoutJSON) => {
+    setCurrentLayout(layout)
+    persistenceManager.saveCurrentLayout(layout)
+  }, []);
 
-    if (json) {
-      // Persist layouts
-      const generatedId = persistenceManager.createLayout(metadata, json);
+  const saveLayout = useCallback((metadata: Omit<LayoutMetadata, "id">) => {
+    persistenceManager.createLayout(metadata, currentLayout).then(generatedId => {
 
-      // Update state
       const newMetadata: LayoutMetadata = {
         ...metadata,
         id: generatedId
       };
 
       setLayoutMetadata(prev => [...prev, newMetadata]);
-    }
-  }, [])
+    })
+  }, [currentLayout])
 
-  // TODO: add loadLayout function
+  const loadLayoutById = useCallback((id: string) => {
+    persistenceManager.loadLayout(id).then((layoutJson) => {
+      setCurrentLayout(prev => ({
+        ...prev,
+        children: [...(prev.children || []), ...(layoutJson.children || [])]
+      }))
```

Comment:
Not quite sure what's happening here. Why is `prev` separate to `layoutJson.children`? What's `prev`, `layoutJson`, and `children`?

Reply:
in this context prev represents the previous version of the state that is being updated here so here prev is the currentLayout state prior the change